### PR TITLE
MCKIN-10776 Get completion in iframe when gating is applied

### PR DESCRIPTION
**Pull request by moeez96**
**Jira ticket:** https://edx-wiki.atlassian.net/browse/MCKIN-10776

**Description:** 
A button is introduced in iframe when gating is applied for user to click on and check completion. If content is completed, gating will be disabled for that xblock and the right arrow will be enabled if no other gating is applied.

**For Reviewer:**
Things to check are:
1. Completion button is only visible in iframe when gating is applied on it and it is incomplete
2. When complete button is clicked iframe is restarted and feedback is shown to the user below.
3. If content is complete but other gating is applied on the button, the right button should stay disabled.
4. Other ticket AC's

### CODE REVIEW

File: scormxblock/scormxblock.py [hunk 1]
```diff
@@ -470,6 +472,14 @@ def set_raw_scorm_status(self, request, suffix=''):
         # TODO: handle errors
         return Response(json.dumps(self.raw_scorm_status), content_type='application/json', charset='UTF-8')
 
+    @XBlock.handler
+    def get_scorm_completion(self, request, suffix=''):
+        completion = {'completion': self.scorm_progress or 0}
```
Comment (jawad-khan) [inline scormxblock/scormxblock.py]: No need for this or, since by default scorm_progress is 0.

File: scormxblock/static/js/src/scormxblock.js [hunk 2]
```diff
@@ -78,35 +78,112 @@ function ScormXBlock_${block_id}(runtime, element) {
   $(function ($) {
     API = new SCORM_API();
     console.log("Initial SCORM data...");
+    const completedFeedbackText = gettext('Content is complete, please continue.')
```
Comment (jawad-khan) [inline scormxblock/static/js/src/scormxblock.js]: semi colons missing.

File: scormxblock/scormxblock.py [hunk 3]
```diff
@@ -236,10 +236,11 @@ def student_view(self, context=None, authoring=False):
         if not authoring:
             get_url = '{}://{}{}'.format(scheme, lms_base, self.runtime.handler_url(self, "get_raw_scorm_status"))
             set_url = '{}://{}{}'.format(scheme, lms_base, self.runtime.handler_url(self, "set_raw_scorm_status"))
+            get_completion = '{}://{}{}'.format(scheme, lms_base, self.runtime.handler_url(self, "get_scorm_completion"))
         # PreviewModuleSystem (runtime Mixin from Studio) won't have a hostname
         else:
             # we don't want to get/set SCORM status from preview
-            get_url = set_url = '#'
+            get_url = set_url = get_completion = '#'
```
Comment (jawad-khan) [inline scormxblock/scormxblock.py]: get_completion sounds like a function name. can we add url to name?

File: scormxblock/scormxblock.py (hunk 1 above)
Comment (moeez96) [reply scormxblock/scormxblock.py]: Just an extra layer of type checking

**Pull request merged**